src/com/ucm/dasi/catan/board/CatanBoard.java

```diff
@@ -0,0 +1,103 @@
+package com.ucm.dasi.catan.board;
+
+import com.ucm.dasi.catan.board.exception.InvalidBoardDimensionsException;
+import com.ucm.dasi.catan.board.exception.InvalidBoardElementException;
+
+public class CatanBoard implements ICatanBoard {
+	
+	protected int terrainWidth;
+	
+	protected int terrainHeight;
+	
+	protected int width;
+	
+	protected int height;
+	
+	protected IBoardElement[][] elements;
+	
+	public CatanBoard(int width, int height, IBoardElement[][] elements) throws InvalidBoardDimensionsException, InvalidBoardElementException {
+		setDimensions(width, height);
```

Comment:
I think it is better to unify these two methods, under the name of init or something like that, for two reasons. The first is because both share an exception and the second is because it is self-contained.

Reply:
`setDimensions` and `setElements` can be called in an `init` method, but it's not a good idea to mix them in a method in order to respect the single responsability principle.

Reply:
Is it possible to call the set Dimensions method without calling the method setElements 
or vice versa? I mean by this that it is actually the same functionality divided in two parts. That's my point.

Reply:
It's not important, they are private methods. I would buy your argument if those methods were public, but as long as they are private we control the code flow.

That approach is taken in order to structure the code. It's easier to understand two small functions instead of a big one. The outcome of this is to try to write "small" functions with a single responsability instead of a big one function.

The responsability of the `setElements` method is to initialize the `elements` attribute. The responsability of the `setDimensions` method is to initialize the dimension atributes. We can find a big responsability to justify the mix process, but it would be a bad practice.

Reply:
Rather, does it make sense to call one without the other?